linetools/spectra/xspectrum1d.py

```diff
@@ -182,6 +184,12 @@ def __init__(self, wave, flux, sig=None, co=None, units=None, select=0,
             if wave.shape[0] != sig.shape[0]:
                 raise IOError("Shape of wave and sig vectors must be identical")
             self.data['sig'] = np.reshape(sig, (self.nspec, self.npix))
+            if mask_edges:
+                for kk in range(self.nspec):
+                    sigval = np.where(self.data['sig'][kk] > 0.)[0]
+                    for key in self.data.dtype.names:
+                        self.data[key][kk][0:sigval[0]].mask = True
+                        self.data[key][kk][sigval[-1]+1:].mask = True
```

Comment:
This will only work for masking out edges, I think.

How about define a boolean array: `valid_sigma = self.data['sig'][kk] > 0.`, and then use the negation of this as the mask? e.g. `self.data[key][kk][0:sigval[0]].mask = ~valid_sigma`